### ISSUE TITLE
Semantic version -build#  in build doesn't work with dependency resolver

### ISSUE THREAD

**Issue by ekkizogloy** [iotile/coretools#533]
After building `lib_controller` with a semantic version `3.7.5-build101` running `iotile depends update` does not locate the dependency. 

<br>

`module_settings.json` file for lib_controller
```
 "version": "3.7.5-build101",
```
<br>
`module_settings.json` for con_btc1

```
"iotile_standard_library/lib_controller,^3.7.5": ["include_directories", "libcontroller_nrf52832.a", "tilebus_definitions"],   
```

`iotile depends update` output:
```
>iotile depends update
Resolving iotile_standard_library/lib_nrf52832: already installed
Resolving iotile_standard_library/lib_controller: not found
Resolving iotile_standard_library/lib_freertos: already installed
Resolving iotile_standard_library/lib_common: already installed
Resolving iotile_standard_library/hw_btc1: already installed
Resolving iotile_standard_library/lib_mbedtls: already installed
```
<br>

When I remove the `-build101` from the version information the dependency resolver works as expected.

**Comment by timburke** [comment]
I think it's a version specification issue.  `3.7.5-build101` is a prerelease build whereas `^3.7.5` is asking for a released firmware >= 3.7.5 and < 4.0.0.  If you want to use the prerelease `lib_controller` you can do:
`^3.7.5-build1` and this should match the prereleased `lib_controller`.  

The default matching operations do not include prerelease firmware versions so that it is safe to mark a release as a prerelease without all released firmware automatically pulling it in (and defeating the goal of making a prerelease).

**Comment by timburke** [comment]
This appears to be working as intended, please reopen if needed.
